docs/extensions/referenced-resources.md

```diff
@@ -0,0 +1,29 @@
+# Referenced Resources
+
+The Shoot resource can include a list of resources (usually secrets) that can be referenced by name in extension `providerConfig` and other Shoot sections, for example:
+
+```yaml
+kind: Shoot
+apiVersion: core.gardener.cloud/v1beta1
+metadata:
+  name: crazy-botany
+  namespace: garden-dev
+  ...
+spec:
+  ...
+  extensions:
+  - type: foobar
+    providerConfig:
+      apiVersion: foobar.extensions.gardener.cloud/v1alpha1
+      kind: FooBarConfig
+      foo: bar
+      secretRef: foobar-secret
+  resources:
+  - name: foobar-secret
+    resourceRef:
+      apiVersion: v1
+      kind: Secret
+      name: my-foobar-secret
+```
+
+Gardener expects to find these referenced resources in the project namespace (e.g. `garden-dev`) and will copy them to the Shoot namespace in the Seed cluster when reconciling a Shoot. Extension controllers can resolve the references to these resources by accessing the Shoot via the `Cluster` resource.
```

Comment:
Please adapt the docs according to your changes with the prefix and the helper function (`GetObjectByReference`).

Reply:
Done